app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.test.ts

```diff
@@ -0,0 +1,191 @@
+      ).toHaveBeenCalledWith(mockMetaMetricsId);
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: mockTaskId,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        metaMetricsDataDeletionStatus: 'UNKNOWN',
+        participateInMetricsDuringDeletion: false,
+      });
+    });
+
+    it('fails to creates a data deletion task when user has never participating in metrics tracking', async () => {
+      const { controller } = setupController({
+        options: {
+          getMetaMetricsId: jest.fn().mockReturnValue(null),
+          getParticipateInMetrics: jest.fn().mockReturnValue(null),
+        },
+      });
+      await expect(
+        controller.createMetaMetricsDataDeletionTask(),
+      ).rejects.toThrow();
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: null,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        participateInMetricsDuringDeletion: null,
+      });
+    });
+  });
+  describe('updateDataDeletionTaskStatus', () => {
+    it('fetches and stores status of the delete regulation using delete regulation ID', async () => {
+      const mockMetaMetricsId = 'mockId';
+      const mockTaskId = 'mockTaskId';
+      const { controller, dataDeletionService } = setupController({
+        options: {
+          getMetaMetricsId: jest.fn().mockReturnValue(mockMetaMetricsId),
+          getParticipateInMetrics: jest.fn().mockReturnValue(true),
+        },
+        dataDeletionService: {
+          createDataDeletionRegulationTask: jest
+            .fn()
+            .mockResolvedValue(mockTaskId),
+          fetchDeletionRegulationStatus: jest.fn().mockResolvedValue('UNKNOWN'),
+        },
+      });
+      await controller.createMetaMetricsDataDeletionTask();
+      await controller.updateDataDeletionTaskStatus();
+      expect(
+        dataDeletionService.fetchDeletionRegulationStatus,
+      ).toHaveBeenCalledTimes(2);
+      expect(
+        dataDeletionService.fetchDeletionRegulationStatus,
+      ).toHaveBeenCalledWith(mockTaskId);
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: mockTaskId,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        metaMetricsDataDeletionStatus: 'UNKNOWN',
+        participateInMetricsDuringDeletion: true,
+      });
+    });
+  });
+});
+
+/**
+ * Setup a test controller instance.
+ *
+ * @param options - Setup options.
+ * @param options.options - Controller constructor options.
+ * @param options.dataDeletionService - A partial mock of the DataDeletionService.
+ * @returns The test controller, a messenger instance, and related mocks.
+ */
+function setupController({
+  options,
+  dataDeletionService,
+}: {
+  options?: Partial<
+    ConstructorParameters<typeof MetaMetricsDataDeletionController>[0]
+  >;
+  dataDeletionService?: Partial<
+    ConstructorParameters<
+      typeof MetaMetricsDataDeletionController
+    >[0]['dataDeletionService']
+  >;
```

Comment:
nit: Is there a reason we have the `dataDeletionService` argument outside of the options bag? It might be confusing to have the controller constructor options scattered in different locations.

Also, it would be helpful to provide JSDoc entries for the constructor options.

```suggestion
 * @param options - Setup options.
 * @param options.options - Controller constructor options.
 * @param options.options.dataDeletionService - A partial mock of the DataDeletionService.
 * @param options.options.getMetaMetricsId - A mock function that returns a mocked MetaMetrics ID.
 * @param options.options.getParticipateInMetrics - A mocked implementation of the `getParticipateInMetrics` method.
 * @returns The test controller, a messenger instance, and related mocks.
 */
function setupController({
  options,
}: {
  options?: Partial<
    ConstructorParameters<typeof MetaMetricsDataDeletionController>[0]
  >;
```